File: gitdot-web/packages/api/src/endpoint/review/merge-review.ts
import { z } from "zod";
import { ReviewResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const MergeReviewRequest = z.object({});
export type MergeReviewRequest = z.infer<typeof MergeReviewRequest>;

export const MergeReviewResponse = ReviewResource;
export type MergeReviewResponse = z.infer<typeof MergeReviewResponse>;

export const MergeReview = {
  path: "/repository/{owner}/{repo}/review/{number}/merge",
  method: "POST",
  request: MergeReviewRequest,
  response: MergeReviewResponse,
} as const satisfies Endpoint;
export type MergeReview = typeof MergeReview;

File: gitdot-api/src/endpoint/review/merge_review.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::review::ReviewResource};

pub struct MergeReview;

impl Endpoint for MergeReview {
    const PATH: &'static str = "/repository/{owner}/{repo}/review/{number}/merge";
    const METHOD: http::Method = http::Method::POST;

    type Request = MergeReviewRequest;
    type Response = MergeReviewResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct MergeReviewRequest {}

pub type MergeReviewResponse = ReviewResource;
